### PR TITLE
INTMDB-285: Fix org_invitations issue

### PR DESCRIPTION
**Pull request by abner-dou**
## Description

Fixed issue in resource org_invitations when accepting organization invitation that causes 404 error, also some error returns was updated

Link to any related issue(s):[INTMDB-285](https://jira.mongodb.org/browse/INTMDB-285), [#636](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/636)

## Type of change:

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update
- [X] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [X] I have added any necessary documentation (if appropriate)
- [X] I have run make fmt and formatted my code

## Further comments
TESTS:
-----------------------------------------------------
--- PASS: TestAccResourceMongoDBAtlasOrgInvitation_importBasic (4.81s)
PASS
coverage: 3.1% of statements
ok      github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas 7.330s  coverage: 3.1% of statements
?       github.com/mongodb/terraform-provider-mongodbatlas/version      [no test files]

-----------------------------------------------------
--- PASS: TestAccResourceMongoDBAtlasOrgInvitation_basic (6.47s)
PASS
coverage: 3.0% of statements
ok      github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas 8.935s  coverage: 3.0% of statements
?       github.com/mongodb/terraform-provider-mongodbatlas/version      [no test files]

### CODE REVIEW

File: website/docs/r/org_invitation.html.markdown [hunk 1]
```diff
@@ -69,10 +69,14 @@ In addition to the arguments, this resource exports the following attributes:
 * `invitation_id` - Unique 24-hexadecimal digit string that identifies the invitation in Atlas.
 * `inviter_username` - Atlas user who invited `username` to the organization.
 
+> **NOTE**: If the user declines the invitation or accepts and leave the organization, terraform provider will send again the invitation if you don't remove the invitation from the resource definition.
```
Comment (themantissa) [inline website/docs/r/org_invitation.html.markdown]:
```suggestion
> **NOTE**: If the user declines the invitation or accepts the invitation and then leaves the organization, the  provider will re-add the invitation the invitation if the resource definition is not removed from the Terraform configuration. 
```

File: mongodbatlas/resource_mongodbatlas_org_invitation.go [hunk 2]
```diff
@@ -245,3 +247,18 @@ func splitOrgInvitationImportID(id string) (orgID, username string, err error) {
...
 }
+
+func validateOrgInvitationAlreadyAccepted(ctx context.Context, conn *MongoDBClient, username, orgID string) (bool, error) {
+	user, _, err := conn.Atlas.AtlasUsers.GetByName(ctx, username)
+	if err != nil {
+		return false, err
+	}
+
+	for _, role := range user.Roles {
+		if role.OrgID == orgID {
```
Comment (themantissa) [inline mongodbatlas/resource_mongodbatlas_org_invitation.go:258]: Our slack conversation dropped off but I think we should explain to the user what happened somehow. Basically the user will have no way to know how we handle when the user accepts.  We should at least document this.  That when the user accepts the invitations the provider will validate that they are now a member of the organization.  That when that happens we will remove the invitation from state and suggest the user removes the invitation from the Terraform configuration, unless they want the user immediately re-invited upon leaving.

**Review by themantissa — CHANGES_REQUESTED** [review]
Small docs comment and also a suggestion to at least document what we do when an org invite is accepted so users know.

**Review by thetonymaster — APPROVED** [review]

File: mongodbatlas/resource_mongodbatlas_org_invitation.go (hunk 2 above)
Comment (abner-dou) [reply mongodbatlas/resource_mongodbatlas_org_invitation.go:258]: I added more documentation to the org_invitations: 

> **NOTE**: The provider also will validate 3 cases:
>* If the user haven't accepted the invitation, nothing happens.
>* If the user already belongs to the organization, the provider will remove the invitation from the state, but it's necessary to remove from the terraform resource definition.
>* If the user is not a member of the organization or rejected the invitation, the provider will try to send again the invitation (unless the invitation get removed from the terraform resource definition).

File: website/docs/r/org_invitation.html.markdown (hunk 1 above)
Comment (abner-dou) [reply website/docs/r/org_invitation.html.markdown]: done!

File: website/docs/r/org_invitation.html.markdown [hunk 3]
```diff
@@ -69,10 +69,18 @@ In addition to the arguments, this resource exports the following attributes:
 * `invitation_id` - Unique 24-hexadecimal digit string that identifies the invitation in Atlas.
 * `inviter_username` - Atlas user who invited `username` to the organization.
 
+> **NOTE**: If the user declines the invitation or accepts the invitation and then leaves the organization, the  provider will re-add the invitation the invitation if the resource definition is not removed from the Terraform configuration.
+
+> **NOTE**: The provider also will validate 3 cases:
+>* If the user haven't accepted the invitation, nothing happens.
+>* If the user already belongs to the organization, the provider will remove the invitation from the state, but it's necessary to remove from the terraform resource configuration.
+>* If the user is not a member of the organization or rejected the invitation, the provider will try to send again the invitation (unless the invitation get removed from the terraform resource configuration). 
+
```
Comment (themantissa) [inline website/docs/r/org_invitation.html.markdown]: In general I think this is headed in the right direction but might still be a bit confusing for users.  I just ran through an invitation process and a user cannot reject an invitation.  They can not accept but only the sender of the invitation can delete it.   Taking your idea and just editing it a bit to what I hope will be clearer: 

```suggestion
> **NOTE**: Possible provider behavior depending on the invitee's action:
>* If the user has not yet accepted the invitation, the provider leaves the invitation as is.
>* If the user has accepted the invitation and is now an organization member, the provider will remove the invitation from the Terraform state.  The invitation must then be removed from the Terraform resource configuration.
>* If the user accepts the invitation and then leaves the organization, the provider will re-add the invitation if the resource definition is not removed from the Terraform configuration.

```

**Review by themantissa — CHANGES_REQUESTED** [review]
Your approach to making the behavior clear should hopefully work.  Just adjusted it based on a quick test.  Thanks!

File: website/docs/r/org_invitation.html.markdown (hunk 3 above)
Comment (abner-dou) [reply website/docs/r/org_invitation.html.markdown]: Done! and thank you for the suggestion :)

**Review by themantissa — APPROVED** [review]
LGTM, thank you!

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**